src/vm/nativefieldcategory.h

```diff
@@ -1,29 +1,23 @@
-enum NativeFieldSubcategory
+enum NativeFieldSubcategory : short
```

Comment:
This is a really interesting enum. I am not a fan of this kind of splitting concepts across this kind of boundary since it is one step away from doing this entirely with macros. What logic are you trying to handle here?

Reply:
The idea with separating out the subcategories is that some concepts are relevant to groups of the native field categories (like the FLOAT category is for R4, R8, and DATE categories and the NESTED group for NESTED_TYPE and IN_PLACE_ARRAY categories).

The groups can be used in different situations to collapse the category checking when possible (i.e. checking if a type is an HFA type requires knowing if a type has a nested type, and checking if a type is an SSE type in the SystemV ABI requires knowing if a type is a FLOAT).

Basically it's supposed to be a set of more abstract flags to group the categories below. I'm thinking I should probably swap the naming.